pkg/fake/iamapi.go

```diff
@@ -39,126 +40,169 @@ type IAMAPIBehavior struct {
 	AddRoleToInstanceProfileBehavior      MockedFunction[iam.AddRoleToInstanceProfileInput, iam.AddRoleToInstanceProfileOutput]
 	TagInstanceProfileBehavior            MockedFunction[iam.TagInstanceProfileInput, iam.TagInstanceProfileOutput]
 	RemoveRoleFromInstanceProfileBehavior MockedFunction[iam.RemoveRoleFromInstanceProfileInput, iam.RemoveRoleFromInstanceProfileOutput]
+	UntagInstanceProfileBehavior          MockedFunction[iam.UntagInstanceProfileInput, iam.UntagInstanceProfileOutput]
 }
 
 type IAMAPI struct {
 	sync.Mutex
 
-	iamiface.IAMAPI
+	sdk.IAMAPI
 	IAMAPIBehavior
 
-	InstanceProfiles map[string]*iam.InstanceProfile
+	InstanceProfiles map[string]*iamtypes.InstanceProfile
 }
 
 func NewIAMAPI() *IAMAPI {
-	return &IAMAPI{InstanceProfiles: map[string]*iam.InstanceProfile{}}
+	return &IAMAPI{InstanceProfiles: map[string]*iamtypes.InstanceProfile{}}
 }
 
 // Reset must be called between tests otherwise tests will pollute
 // each other.
+func toSliceOfValues[T any](value []T) []T {
+	values := make([]T, len(value))
+	copy(values, value)
+	return values
+}
+
 func (s *IAMAPI) Reset() {
 	s.GetInstanceProfileBehavior.Reset()
 	s.CreateInstanceProfileBehavior.Reset()
 	s.DeleteInstanceProfileBehavior.Reset()
 	s.AddRoleToInstanceProfileBehavior.Reset()
 	s.RemoveRoleFromInstanceProfileBehavior.Reset()
-	s.InstanceProfiles = map[string]*iam.InstanceProfile{}
+	s.InstanceProfiles = map[string]*iamtypes.InstanceProfile{}
 }
 
-func (s *IAMAPI) GetInstanceProfileWithContext(_ context.Context, input *iam.GetInstanceProfileInput, _ ...request.Option) (*iam.GetInstanceProfileOutput, error) {
+func (s *IAMAPI) GetInstanceProfile(ctx context.Context, input *iam.GetInstanceProfileInput, _ ...func(*iam.Options)) (*iam.GetInstanceProfileOutput, error) {
```

Comment:
nit 

It does not seem like we use the context. So we can leave these as they were

```suggestion
func (s *IAMAPI) GetInstanceProfile(_ context.Context, input *iam.GetInstanceProfileInput, _ ...func(*iam.Options)) (*iam.GetInstanceProfileOutput, error) {
```